subsys/bluetooth/controller/ll_sw/isoal.c

```diff
@@ -1276,6 +1494,31 @@ static isoal_status_t isoal_tx_unframed_produce(struct isoal_source *source,
 		pp->payload_number = MAX(pp->payload_number,
 			(tx_sdu->target_event * session->burst_number));
 
+		/* Get actual event for this payload number */
+		actual_event = pp->payload_number / session->burst_number;
+
+		/* Get group reference point for this PDU based on the actual
+		 * event being set. This might introduce some errors as the
+		 * group refernce point for future events could drift. However
+		 * as the time offset calculation requires an absolute value,
+		 * this seems to be the best candidate.
+		 */
+		if (actual_event > tx_sdu->target_event) {
+			actual_grp_ref_point = tx_sdu->grp_ref_point +
+				((actual_event - tx_sdu->target_event) * session->iso_interval *
+					ISO_INT_UNIT_US);
```

Comment:
Any risk that they may overflow? Looks like we are adding a uint64_t times `session->iso_interval * ISO_INT_UNIT_US)` to a `uint32_t` without any checks of whether the result can be stored.

Reply:
First off the event count is only 39 bits of the 64. 64 bits were used for convenience.

Even so, yes this will overflow but that overflow is expected. The reason for that is that the `grp_ref_point` here is derived from the controller's clock and that is based off a 32-bit value. In Oticon's case a tick is 1/16 of a microsecond so we effectively have only 28 bits for the microsecond range. That is also going to overflow and wrap around. Similarly the time-stamp field according to the specs is also only 32-bits. The important thing here is that the derived reference point and any calculations based on those are accurate across the wrap around and comparable to the controller's clock in the microsecond range.

Reply:
Alright. If an overflow is possible by design, I would suggest to add a small comment about that, as that can be a hard issue to debug otherwise :)

Reply:
I did find a problem in the time-offset calculation in the framed case that would have been affected if the reference point was to wrap around, so I changed the intermediate reference point to 64-bits. Thanks for pointing this out.